### PR TITLE
Refactor tests for punisher.

### PR DESCRIPTION
**Pull request by drvinceknight**
Contributes towards #884

### CODE REVIEW

File: axelrod/tests/strategies/test_punisher.py [hunk 1]
```diff
@@ -22,30 +22,37 @@ class TestPunisher(TestPlayer):
 
     def test_init(self):
         """Tests for the __init__ method."""
-        P1 = axelrod.Punisher()
-        self.assertEqual(P1.history, [])
-        self.assertEqual(P1.mem_length, 1)
-        self.assertEqual(P1.grudged, False)
-        self.assertEqual(P1.grudge_memory, 1)
+        player = axelrod.Punisher()
+        self.assertEqual(player.history, [])
```
Comment (marcharper) [inline axelrod/tests/strategies/test_punisher.py]: This line is probably not necessary, or is better as a generic player test in `test_player.py`.

File: axelrod/tests/strategies/test_punisher.py [hunk 2]
```diff
@@ -64,34 +71,37 @@ class TestInversePunisher(TestPlayer):
 
     def test_init(self):
         """Tests for the __init__ method."""
-        P1 = axelrod.InversePunisher()
-        self.assertEqual(P1.history, [])
-        self.assertEqual(P1.mem_length, 1)
-        self.assertEqual(P1.grudged, False)
-        self.assertEqual(P1.grudge_memory, 1)
+        player = axelrod.InversePunisher()
+        self.assertEqual(player.history, [])
```
Comment (marcharper) [inline axelrod/tests/strategies/test_punisher.py]: Same comment

File: axelrod/tests/strategies/test_punisher.py (hunk 1 above)
Comment (drvinceknight) [inline axelrod/tests/strategies/test_punisher.py]: Yeah good call. I'll clean this up now :+1:

Comment (drvinceknight) [inline axelrod/tests/strategies/test_punisher.py]: Just checked and it already is a generic player test :)

**Pull request merged**